### PR TITLE
[Merged by Bors] - feat: Prove that the integral of a function composed with the absolute value is twice the integral over the positive

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/Lebesgue/Integral.lean [hunk 1]
```diff
@@ -104,3 +104,30 @@ theorem integral_comp_neg_Ioi {E : Type*} [NormedAddCommGroup E] [NormedSpace 
   rw [← neg_neg c, ← integral_comp_neg_Iic]
   simp only [neg_neg]
 #align integral_comp_neg_Ioi integral_comp_neg_Ioi
+
+theorem integral_comp_abs {f : ℝ → ℝ} :
+    ∫ x, f |x| = 2 * ∫ x in Ioi (0:ℝ), f x := by
+  have eq : ∫ (x : ℝ) in Ioi 0, f |x| = ∫ (x : ℝ) in Ioi 0, f x := by
+    refine set_integral_congr measurableSet_Ioi (fun _ hx => ?_)
+    rw [abs_eq_self.mpr (le_of_lt (by exact hx))]
+  obtain hf | hf := em (IntegrableOn (fun x => f |x|) (Ioi 0))
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/Lebesgue/Integral.lean:113]: Could you use instead
```suggestion
  by_cases IntegrableOn (fun x => f |x|) (Ioi 0)
```
It looks more readable to me.

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/MeasureTheory/Measure/Lebesgue/Integral.lean (hunk 1 above)
Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Lebesgue/Integral.lean]: You are right. I don't know why I complicated things so much here :) Thanks

**Pull request closed without merging**